compiler/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -260,6 +262,11 @@ impl<'a> From<&'a TypeCheckError> for Diagnostic {
                 let msg = format!("Function expects {expected} parameter{empty_or_s} but {found} {was_or_were} given");
                 Diagnostic::simple_error(msg, String::new(), *span)
             }
+            TypeCheckError::AssertionParameterCountMismatch { expected1, expected2, found, span } => {
+                let was_or_were = if *found == 1 { "was" } else { "were" };
+                let msg = format!("Function expects {expected1} or {expected2} parameters but {found} {was_or_were} given");
```

Comment:
I wonder if we should explicitly say assert or assert_eq here. Since they aren't really functions since they're not first class like the others are. I suppose if we passed the ConstrainKind we also wouldn't need to pass the expected argument count anymore either. It could be a method on ConstrainKind.

Reply:
Done! I hesitated between that method returning somehow the min and max values, but ended up just returning the minimum arguments count and assuming the caller would know the maximum is that plus one (maybe it's clearer).